geomesa-core/src/test/scala/org/locationtech/geomesa/core/index/AttributeIndexStrategyTest.scala

```diff
@@ -129,6 +130,29 @@ class AttributeIndexStrategyTest extends Specification {
       success
     }
 
+    "partition a query by selecting the best filter" >> {
```

Comment:
It may be worth adding a few more unit tests like this that demonstrate this working like the cases you added in FilterHelperTest but this could be considered redundant.